### DISCUSSION TITLE
Inhibit notifications when in a fullscreen application?

### DISCUSSION THREAD

**Discussion by poperigby** [ErikReider/SwayNotificationCenter#432] [Q&A]
I'd like to inhibit notifications when I'm viewing a fullscreen application in Sway, because notifications popping up over game windows is causing stutters. How can I do this?

**Comment by cyrneko** [comment]
Anyone have a solution for this? :upside_down_face:

**Comment by MangelMaxime** [comment]
This script doesn't reply exactly to your needs but could give you an idea on how to set this up.

It listen to a socket for `Hyperland` and when it detects that I am sharing my screen to will inhibit notifications. 

If you find a way in sway to detect when have a fullscreen application, you should be able to adapt this script.

```bash
#!/bin/sh

toggleNotifications () {
    # Extract the first number after >>
    state=$(echo "$1" | cut -d'>' -f3 | cut -d',' -f1)

    if [ "$state" -eq 1 ]; then
        echo "🎥 Screencast started"
        swaync-client --inhibitor-add "xdg-desktop-portal-wlr"
        swaync-client --dnd-on
    else
        echo "🛑 Screencast stopped"
        swaync-client --inhibitor-remove "xdg-desktop-portal-wlr"
        swaync-client --dnd-off
    fi
}

handle() {
  case $1 in
    screencast*) toggleNotifications $1 ;;
  esac
}

socat -U - UNIX-CONNECT:$XDG_RUNTIME_DIR/hypr/$HYPRLAND_INSTANCE_SIGNATURE/.socket2.sock | while read -r line; do handle "$line"; done
```
